From: Leon Plickat <leonhenrik.plickat@stud.uni-goettingen.de>
To: <~sircmpwn/wio@lists.sr.ht>
CC: Leon Plickat <leonhenrik.plickat@stud.uni-goettingen.de>
Subject: [PATCH 3/3] Selection when reshaping view is now representative
Date: Wed, 12 Jun 2019 01:15:12 +0200
Message-ID: <20190611231512.4173-3-leonhenrik.plickat@stud.uni-goettingen.de>
X-Mailer: git-send-email 2.22.0
In-Reply-To: <20190611231512.4173-1-leonhenrik.plickat@stud.uni-goettingen.de>
References: <20190611231512.4173-1-leonhenrik.plickat@stud.uni-goettingen.de>
MIME-Version: 1.0
Content-Type: text/plain
X-Originating-IP: [134.76.63.32]
X-ClientProxiedBy: EXCMBX-02.um.gwdg.de (134.76.9.217) To EXCMBX-04.um.gwdg.de
 (134.76.9.219)
X-Virus-Scanned: (clean) by clamav
Content-Transfer-Encoding: quoted-printable

The view was moved to far in the x and y direction compared to the
selection outline by exactly the size of the window border.
---
 input.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/input.c b/input.c
index 99d1b4b..b4d8d13 100644
--- a/input.c
+++ b/input.c
@@ -332,8 +332,8 @@ static void handle_button_internal(
 		}
 		break;
 	case INPUT_STATE_RESIZE_END:
-		x1 =3D server->interactive.sx, x2 =3D server->cursor->x;
-		y1 =3D server->interactive.sy, y2 =3D server->cursor->y;
+		x1 =3D server->interactive.sx - window_border, x2 =3D server->cursor->=
x - window_border;
+		y1 =3D server->interactive.sy - window_border, y2 =3D server->cursor->=
y - window_border;
 		if (x2 < x1) {
 			int _ =3D x1;
 			x1 =3D x2;
--=20